source/develop/release-management/features/network/isolated-ports.md

```diff
@@ -0,0 +1,245 @@
+      <model type="virtio"/>
+      <link state="up"/>
+      <source bridge="isolated_0"/>
+      <port isolated="yes"/>
+      <alias name="ua-6e2e236c-84e2-4601-83e4-2b1c66da46c9"/>
+      <mac address="56:6f:74:24:00:02"/>
+      <mtu size="1500"/>
+      <filterref filter="vdsm-no-mac-spoofing"/>
+      <bandwidth/>
+    </interface>
+```
+
+### Ensuring the Prerequisites
+
+Some ideas on how to ensure that the VMs are scheduled only to compatible oVirt
+hosts are listed in the discussion of [open issues](#open-issues).
+
+## Installation/Upgrade
+
+The feature will not affect new or existing installations.
+
+## User work-flows
+
+1. The user ensures that there is an unused Ethernet port on all hosts in the
+   relevant cluster.
+2. Ensure that the switch prevents communication between the switch ports
+   connected to the free Ethernet ports.
+3. Create a new logical network and enables the attributes "VM Network" and
+   "isolated".
+4. Attach the new logical network to the unused Ethernet port on all relevant
+   hosts in the cluster.
+5. Attache the new logical network to a vNIC and run the VM.
+6. Verify that two VMs on the same host cannot communicate with each other.
+7. Verify that two VMs on separate hosts cannot communicate with each other.
+8. Verify that these VMs can communicate with the external (northbound, uplink)
+   network.
+
+## Testing
+
+to test the feature in oVirt, it is sufficient to ensure that two VMs on the
+same host are able to communicate with the gateway, but not with each other.
+For the single host scenario, it is not necessary to configure the switch.
+
+## Open Issues
+
+The feature depends on libvirt from RHEL AV 8.3, because this version of
+libvirt introduced support support for [isolated ports][13].
+This way the feature will not work if a VM is scheduled to a host with a lower
+version of libvirt.
+Possible solutions:
+
+* We introduce a new cluster level, which is clean, but requires additional
+  effort.
+* We introduce a new datacenter level, which is mostly clean, but requires
+  additional effort.
+* The engine ensures that related networks are only connected to a host with an
+  appropriate version of libvirt.
+* The engine ensures that the related networks are only connected to VMs with a
+  machine type that requires a RHEL 8.3 host.
+* You have to manually enable the feature using engine-config, and only if you
+  are sure that all hosts have an appropriate libvirt version.
+* The user has additional effort required.
+* A combination of multiple solutions. We could create a cluster version
+  dependent configuration value, e.g. "IsPortIsolationSupported", and set this
+  to false for cluster levels up to 4.4 . If a user is sure that she wants to
+  use this feature already, because all the oVirt hosts use AV 8.3, she could
+  enable it for cluster level 4.4. The engine would allow only to connect
+  adding relevant networks to VMs, if IsPortIsolationSupported is true for the
```

Comment:
I think that until we have a new compatibility version the last part makes the most sense.
Would trying to run a VM with an isolated port via libvirt that doesn't support this end up in failure or the VM would run without that port being isolated? I don't remember how libvirt reacts to unsupported elements in the xml...

Reply:
sometimes it drops them silently, sometimes it aborts vm

Reply:
and we're not suggesting that, right, we're just going to do it this way. disable for 4.4, once we introduce 4.5 we will enable it there.

Reply:
To my experience this depends, I don't know the generic rule.
In this case, libvirt ignores silently and boots the VM on a libvirt version without the feature.

I am going to remove the open issues section and move this approach to "Ensuring the Prerequisites"

Reply:
ok, so worth documenting that - it can be a source of confusion I suppose